### PR TITLE
Feat: allow customizing ramscoop per system and turn universal back on by default

### PR DESCRIPTION
**Pull request by Hurleveur**
**Feature:** This PR addresses https://github.com/endless-sky/endless-sky/issues/7736, by giving more flexibility for ramscoops as well as turning the universal back on by default.

## Feature Details
Allows modifications of the way system ramscoops work:
- disabling (and re-enabling by event) of the universal ramscoop per system (defaulted back to on)
- applying a multiplier on the ramscoop provided by the ship
- adding/removing a raw value on the amount of fuel generated per tick

## UI Screenshots
n/a

## Usage Examples
Thinking of using this for all systems in the Ember Wastes to make one need to be prepared when going in those.
to do that:
```
system "any system"
    ramscoop
        universal 0

# to add it back
event "add it back"
    system "any system"
        ramscoop
              universal 1
```

Now as for the other attributes, here is an example:
```
system "Sol"
	ramscoop
		universal 0
		multiplier 0.9
		addend -0.04
	remove ramscoop # Wipe all ramscoop settings back to default
```

The universal ramscoop is just a base ramscoop value for all systems, the multiplier is applied on top of said universal and the amount of ramscoop a ship has (square-rooted, as always)
Then the addend is applied on top of that, and can be used to have a strong solar wind without much ramscoop available, or the opposite.

## Testing Done
Not yet, tell me if its needed. 

### Automated Tests Added
n/a

## Performance Impact
n/a

### CODE REVIEW

**Comment by Zitchas** [comment]
This could be useful. If more dangerous parts of the galaxy had the universal ramscoop disabled, that'd satisfy some of my concerns about re-instating universal ramscoop.

**Comment by Hecter94** [comment]
It should be possible to specify in the system definition that universal ramscoop is disabled, rather than having to set it via event. 
It seems like systems will more often have it disabled from the start of the game rather than disabling it during the game.

**Comment by Hurleveur** [comment]
It should indeed be possible to do it since the start *and* by event.

**Comment by Zitchas** [comment]
Yes, they'd predominantly be from the start of the game; but if that's the route we're going instead of universal on/off, then I'll make use of the events to change things as appropriate for the Ember Waste story going forward.

**Comment by Hecter94** [comment]
> It should indeed be possible to do it since the start _and_ by event.

Was that something you wanted to do in this PR or is that outside the scope?

**Comment by Hurleveur** [comment]
I dont do features without enabling event modification and whilst having normal thing working too, especially this simple

**Comment by Amazinite** [comment]
In addition to a per-system on/off switch for the universal ramscoop, could we get a per-system ramscoop multiplier, or perhaps a per-system ramscoop subtractor? Thinking of potential in-universe explanations for why the universal ramscoop might not work while outfit ramscoops still do, imagine the solar wind as a [liquid](https://tvtropes.org/pmwiki/pmwiki.php/Main/SpaceIsAnOcean) and ramscoops as filters, with larger ramscoops being larger/more effective filters. The universal ramscoop would then be analogous to a small filter, one which could be easily clogged if the liquid it were filtering were polluted in some way, but a larger filter would be able to remain unclogged, but perhaps still experience a reduction in efficiency. I feel this concept of polluted solar wind also lends itself well to the Ember Waste.

Syntax wise, that might look like this:
```
system "Sol"
	# The universal ramscoop doesn't work, and all other ramscoops only provide 80% the fuel gen.
	remove "universal ramscoop"
	"ramscoop multiplier" 0.8
```
or with a subtractor, perhaps like this:
```
system "Sol"
	# Subtract 0.042 from the fuel generation/frame of ships using ramscoops. This value is high enough
	# to completely cut off the universal ramscoop from working in any system with a solar wind <= 2.2,
	# meaning you'd need a special star sprite or certain binary star combinations in order to gain any
	# fuel. At that same threshold, a normal human ramscoop would be about 70% efficient, but with
	# a solar wind of 1 it would only be about 33% efficient.
	"ramscoop subtractor" 0.042
```

If we wanted both a multiplier and a subtractor, I'd say have syntax like this:
```
system "Sol"
	ramscoop
		# Or 1 and 0 for now, as true and false value inputs are handled by the gamerules PR
		universal (true | false)
		multiplier 0.9
		subtractor 0.04
	remove ramscoop # Wipe all ramscoop settings back to default
```

**Comment by Hurleveur** [comment]
I did as you suggested except I allowed both negative and positive values for the substractor, which I called modifier.

File: source/System.cpp [hunk 1]
```diff
@@ -176,16 +182,33 @@ void System::Load(const DataNode &node, Set<Planet> &planets)
...
+					universalRamscoop = grand.Value(1);
+				else if(key == "addend" && hasValue)
+					ramscoopAddend = grand.Value(1);
+				else if(key == "multiplier" && hasValue)
+					ramscoopMultiplier = grand.Value(1);
+				else
+					child.PrintTrace("Skipping unrecognized attribute:");
+			}
+		}
 		else if(!hasValue && key != "object")
```
Comment (Hurleveur) [inline source/System.cpp:206]: shouldn't we move object up over this, now?
could be in another PR

File: source/System.cpp [hunk 2]
```diff
@@ -599,6 +622,17 @@ bool System::Hidden() const
 
 
 
+// Return how much ramscoop is generated by this systems, depending the given ship ramscoop value.
+double System::RamscoopFuel(double shipRamscoop, double scale) const
+{
+	// Even if a ship has no ramscoop, it can harvest a tiny bit of fuel by flying close to the star,
+	// provided the system allows it.
+	double universal = 0.05 * scale * universalRamscoop;
+	return SolarWind() * .03 * scale * ramscoopMultiplier * (sqrt(shipRamscoop) + universal) + ramscoopAddend;
```
Comment (Amazinite) [inline source/System.cpp:631]: Don't let a negative added steal fuel.
```suggestion
	return max(0., SolarWind() * .03 * scale * ramscoopMultiplier * (sqrt(shipRamscoop) + universal) + ramscoopAddend);
```
